PG17 compile error and add PG17 ci

I was gearing to add a PG17 cause well stuff is breaking in postgis, so figure stuff might be breaking here too and voila, we have an issue

https://github.com/robe2/pgsql-ogr-fdw/actions/runs/8444294542/job/23129550923

```
Run export PATH=/usr/lib/postgresql/17/bin/:$PATH
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -Werror -fPIC -fvisibility=hidden -I/usr/include/gdal -I. -I./ -I/usr/include/postgresql/17/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2   -c -o ogr_fdw.o ogr_fdw.c
ogr_fdw.c: In function ‘ogrGetForeignPaths’:
ogr_fdw.c:947:26: error: too few arguments to function ‘create_foreignscan_path’
  947 |                  (Path*) create_foreignscan_path(root, baserel,
      |                          ^~~~~~~~~~~~~~~~~~~~~~~
In file included from ogr_fdw.h:45,
                 from ogr_fdw.c:40:
/usr/include/postgresql/17/server/optimizer/pathnode.h:125:21: note: declared here
  125 | extern ForeignPath *create_foreignscan_path(PlannerInfo *root, RelOptInfo *rel,
      |                     ^~~~~~~~~~~~~~~~~~~~~~~
make: *** [<builtin>: ogr_fdw.o] Error 1

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -20,6 +20,7 @@ jobs:
           - { PGVER: 14 }
           - { PGVER: 15 }
           - { PGVER: 16 }
+          - { PGVER: 17 }
 
     runs-on: ubuntu-latest
     steps:

--- a/ogr_fdw.c
+++ b/ogr_fdw.c
@@ -954,6 +954,9 @@ ogrGetForeignPaths(PlannerInfo* root,
 	                 NIL,     /* no pathkeys */
 	                 NULL,    /* no outer rel either */
 	                 NULL  /* no extra plan */
+#if PG_VERSION_NUM >= 170000
+	                 , NIL /* no fdw_restrictinfo list */
+#endif
 #if PG_VERSION_NUM >= 90500
 	                 , NIL /* no fdw_private list */
 #endif

